unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs

```diff
@@ -77,25 +78,20 @@ public void Update()
             }
 
             if (sceneChanged && currentScene != null && currentSceneNumber != currentScene.sceneData.sceneNumber)
-            {
                 currentScene = null;
-            }
+
+            // we get current scene reference
+            currentScene ??= GetCurrentScene(currentSceneNumber, loadedScenes);
 
             // UI not set for current scene yet
             if (isPendingSceneUI)
             {
-                // we get current scene reference
-                currentScene ??= GetCurrentScene(currentSceneNumber, loadedScenes);
-
                 // we apply current scene UI
-                if (currentScene != null)
-                {
-                    if (ApplySceneUI(internalUiContainerComponent, uiDocument, currentScene, isSceneUIEnabled))
-                    {
-                        isPendingSceneUI = false;
-                    }
-                }
+                if (currentScene != null && ApplySceneUI(internalUiContainerComponent, uiDocument, currentScene, isSceneUIEnabled))
+                    isPendingSceneUI = false;
```

Comment:
is this change OK? 
looks like previously `isPendingSceneUI==true` was a requisite to call `currentScene ??= GetCurrentScene(currentSceneNumber, loadedScenes)` but now is being done on every update